packages/trader/src/Modules/Trading/Containers/trade-chart.tsx

```diff
@@ -0,0 +1,200 @@
+            contracts_array={markers_array}
+            bottomWidgets={
+                (is_accumulator || show_digits_stats) && isDesktop() ? getBottomWidgets : props.bottomWidgets
+            }
+            crosshair={is_mobile ? 0 : undefined}
+            crosshairTooltipLeftAllow={560}
+            showLastDigitStats={isDesktop() ? show_digits_stats : false}
+            chartControlsWidgets={null}
+            chartStatusListener={(v: boolean) => setChartStatus(!v)}
+            chartType={chart_type}
+            initialData={{
+                activeSymbols: JSON.parse(JSON.stringify(active_symbols)),
+            }}
+            chartData={{
+                activeSymbols: JSON.parse(JSON.stringify(active_symbols)),
+            }}
+            feedCall={{
+                activeSymbols: false,
+            }}
+            enabledNavigationWidget={isDesktop()}
+            enabledChartFooter={false}
+            id='trade'
+            isMobile={is_mobile}
+            maxTick={is_mobile ? max_ticks : undefined}
+            granularity={show_digits_stats || is_accumulator ? 0 : Number(granularity)}
+            requestAPI={wsSendRequest}
+            requestForget={wsForget}
+            requestForgetStream={wsForgetStream}
+            requestSubscribe={wsSubscribe}
+            settings={settings}
+            should_show_eu_content={should_show_eu_content}
+            allowTickChartTypeOnly={show_digits_stats || is_accumulator}
+            stateChangeListener={chartStateChange}
+            symbol={symbol}
+            topWidgets={is_trade_enabled ? topWidgets : undefined}
+            isConnectionOpened={is_socket_opened}
+            clearChart={false}
+            toolbarWidget={() => {
+                if (is_beta_chart) {
+                    return (
+                        <ToolbarWidgetsBeta
+                            updateChartType={updateChartType}
+                            updateGranularity={updateGranularity}
+                            is_mobile={is_mobile}
+                        />
+                    );
+                }
+                return (
+                    <ToolbarWidgets
+                        updateChartType={updateChartType}
+                        updateGranularity={updateGranularity}
+                        is_mobile={is_mobile}
+                    />
+                );
+            }}
+            importedLayout={chart_layout}
+            onExportLayout={exportLayout}
+            shouldFetchTradingTimes
+            hasAlternativeSource={has_alternative_source}
+            getMarketsOrder={getMarketsOrder}
+            should_zoom_out_on_yaxis={is_accumulator}
+            yAxisMargin={{
+                top: is_mobile ? 76 : 106,
+            }}
+            isLive={true}
+            leftMargin={isDesktop() && is_positions_drawer_on ? 328 : 80}
+            is_beta={is_beta_chart}
+        >
+            {!is_beta_chart &&
+                markers_array.map(marker => {
```

Comment:
mapping markers directly instead of having an extra `ChartMarkers` component we used to have.